lib/filewatch/watch.rb

```diff
@@ -118,13 +118,21 @@ def each(&block)
       synchronized do
         return if @files.empty?
 
+        quitting = false
         file_deleteable = []
         # creates this array just once
         watched_files = @files.values
 
+        quit_proc = lambda do
```

Comment:
I thought about using a method, but my logic is as follows:
- its this method's exit finaliser (i looked v briefly at throw/catch but the syntax looked to be even more confusing)
- the external method would have to have the file_deletable array passed to it so its not really a utility method

I never though about an ensure block - good idea :+1: